src/realm/object-store/sync/sync_session.cpp

```diff
@@ -109,6 +109,13 @@ void SyncSession::become_active()
     // when entering from the Dying state the session will still be bound
     if (!m_session) {
         create_sync_session();
+        // If the session failed to be created (e.g. location was not successfully updated),
+        // then revert back to the inactive state - the error was already passed to the provided
+        // error handler or an exception was thrown
+        if (!m_session) {
+            m_state = State::Inactive;
```

Comment:
I wonder if we should set the state to active after the binding is successful (or not needed if a session exists), then we wouldn't need to revert the state.